src/python/WMCore/WMSpec/StdSpecs/StepChain.py

```diff
@@ -229,7 +230,21 @@ def setupNextSteps(self, task, origArgs):
             parentCmsswStepHelper.keepOutput(parentKeepOutput)
             childKeepOutput = strToBool(taskConf.get('KeepOutput', True))
             childCmsswStepHelper.keepOutput(childKeepOutput)
-            self.setupOutputModules(task, taskConf, currentCmsRun, childKeepOutput)
+            outputMods = self.setupOutputModules(task, taskConf, currentCmsRun, childKeepOutput)
+
+            # set the output dataset parentage
+            if parentKeepOutput and parentStepNumber == "Step1":
+                childStepName = origArgs[parentStepNumber]["StepName"]
+                # TODO how to get the output dataset if first step has KeepOut == True
+                outputModule = outputMods[origArgs[parentStepNumber]['InputFromOutputModule']]
```

Comment:
and here it will break if parentStepNumber is Step1, since there is no InputFromOutputModule attr.